src/Couchbase.Lite.Shared/Query/IndexConfiguration.cs

```diff
@@ -0,0 +1,66 @@
+﻿// 
+// IndexConfiguration.cs
+// 
+// Copyright (c) 2021 Couchbase, Inc All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// 
+
+using LiteCore.Interop;
+using System;
+
+namespace Couchbase.Lite.Internal.Query
+{
+    public abstract class IndexConfiguration
+    {
+        #region Properties
+
+        protected string[] Expression { get; }
+
+        internal C4QueryLanguage QueryLanguage { get; }
```

Comment:
I don't think these are meant to be internal?

Reply:
hmm? Do we need to let the developers know these c4 query language types: C4QueryLanguage.N1QLQuery or C4QueryLanguage.JSONQuery?

Reply:
Oh nevermind I thought these were the public properties that Pasin mentioned (I saw "expression" that looked familiar) but they aren't right?

Reply:
Just confirmed with @pasin That "Expressions" should be public according to the spec.

Reply:
It is "Expressions" (with an S) not "Expression"